internals/templates/routes.js

```diff
@@ -4,22 +4,39 @@
 // They are all wrapped in the App component, which should contain the navbar etc
 // See http://blog.mxstbr.com/2016/01/react-apps-with-pages for more information
 // about the require.ensure code splitting business
+import { injectAsyncReducer } from './store';
+
+function errorLoading(err) {
+  console.error('Dynamic page loading failed', err);
+}
+
+function loadModule(cb) {
+  return (module) => cb(null, module.default);
+}
+
+function loadReducer(store, name) {
+  return (module) => injectAsyncReducer(store, name, module.default);
+}
 
 export default function createRoutes(store) { // eslint-disable-line
   return [
     {
       path: '/',
-      getComponent: function get(location, cb) {
-        require.ensure([], (require) => {
-          cb(null, require('HomePage').default);
-        }, 'HomePage');
+      getComponent(location, cb) {
+        System.import('HomePage/reducer')
```

Comment:
You are using the module loader twice here. So you are generating two promises, one for the reducer, one for the page. But you are calling the callback directly after the second one is resolved. So if the second one resolves faster, the route is loaded before the reducer is attached (unless I am misunderstanding this). Conclusion: You should probably use Promise.all() here.